packages/elements/src/react/common/form/otp.tsx

```diff
@@ -12,29 +13,66 @@ export type OTPInputProps = Exclude<
   autoSubmit?: boolean;
 };
 
-type SelectionRange = readonly [start: number, end: number];
+type SelectionRange = readonly [start: number | null, end: number | null];
 const ZERO: SelectionRange = [0, 0];
 const OUTSIDE: SelectionRange = [-1, -1];
 
 export const OTP_LENGTH_DEFAULT = 6;
 
+/**
+ * If the render prop is provided a custom, segmented input will be rendered. Otherwise, a standard input will be rendered.
+ */
 export const OTPInput = React.forwardRef<HTMLInputElement, OTPInputProps>(function OTPInput(props, ref) {
-  const treatAsSegmented = typeof props.render !== 'undefined';
+  const { render, length, autoSubmit, ...rest } = props;
+  const hasRenderProp = typeof render !== 'undefined';
 
-  if (treatAsSegmented) {
+  if (hasRenderProp) {
     return (
-      // @ts-expect-error -- treatAsSegment guard above ensures props.render is defined
       <OTPInputSegmented
-        {...props}
+        {...rest}
         ref={ref}
+        render={render}
+        length={length}
+        autoSubmit={autoSubmit}
       />
     );
   }
 
   return (
-    <RadixControl
-      {...props}
+    <OTPInputStandard
+      {...rest}
       ref={ref}
+      length={length}
+      autoSubmit={autoSubmit}
+    />
+  );
+});
+
+/**
+ * Standard `<input />` element that receives the same props as the OTPInput component except for the render prop.
+ */
+const OTPInputStandard = React.forwardRef<HTMLInputElement, Omit<OTPInputProps, 'render'>>(function OTPInput(
```

Comment:
I moved the standard component from an inline definition in `OTPInput` to its own component. Mostly to support the `autoSubmit` feature.